pyof/foundation/basic_types.py

```diff
@@ -412,7 +412,7 @@ class BinaryData(GenericType):
     return the size of the instance using Python's :func:`len`.
     """
 
-    def __init__(self, value=b''):  # noqa
+    def __init__(self, value=None):  # noqa
```

Comment:
By "test", I mean unit tests in tests folder. In this case, asserting that the class behaves correctly with a None value.